core/src/main/java/org/mobilitydata/gtfsvalidator/notice/Notice.java

```diff
@@ -40,11 +42,18 @@ public SeverityLevel getSeverityLevel() {
   }
 
   /**
-   * Returns a descriptive type-specific name for this notice.
+   * Returns a descriptive type-specific name for this notice based on the class simple name.
    *
    * @return notice code, e.g., "foreign_key_error".
```

Comment:
Are we going to trim off `_error` too?

I think trimming this off for most notices would work ok (`InvalidRowLengthError`, `DuplicateKeyError`, `MissingRequiredFieldError`) - the only notices that wouldn't work for that I saw are:
* `ForeignKeyError` (maybe this should be `MissingPrimaryKeyError` or `OrphanedForeignKeyError` instead?)
* `IOError`
* `ThreadExecutionError`

Reply:
Should these notice have their own implementation of `getCode`?

Reply:
@barbeau No, we are not going to trim "error". Please see my plan that I have described earlier in https://github.com/MobilityData/gtfs-validator/pull/783#issuecomment-790123561